funcframework/framework.go

```diff
@@ -142,6 +142,10 @@ func Start(port string) error {
 		}
 	}
 
+	if handler == nil {
+		return fmt.Errorf("no matching function found")
+	}
+
 	return http.ListenAndServe(":"+port, handler)
```

Comment:
I suspect the unit tests are failing due to the fact that we are starting up this server and never closing it (probably).

You'll probably need to create a package-global `server` var that we swap out for tests: https://stackoverflow.com/questions/44078189/how-do-i-kill-a-http-server-from-another-function
https://pkg.go.dev/net/http/httptest#NewUnstartedServer


By the way, you can tests all the Golang tests underneath a directory by running `go test ./...`